### ISSUE TITLE
How to encode instructions?

### ISSUE THREAD

**Issue by bhattg** [McGill-NLP/llm2vec#20]
Hi! 

Suppose I want to compute the sematic similarity between bunch of instruction, how do I go about that? Following is in my mind -- 

"Encode the following for semantic search : [instruction]" 

Where the [instruction] will be replaced with "### Instruction: <some task to solve>"

Is this the right way?  Does LLM2Vec recognize the special characters such as "###"?

**Comment by bhattg** [comment]
Any updates on this? I'd appreciate the help of the authors!

**Comment by vaibhavad** [comment]
Hi @bhattg,

Please refer to [Inference](https://github.com/McGill-NLP/llm2vec#inference) guide in README. Specifically the input can be in the form of `[[instruction1, text1], [instruction2, text2]]` or `[text1, text2]`.

You can also refer to an [example](https://github.com/McGill-NLP/llm2vec/blob/main/examples/sts.py) for Semantic Textual Similarity task provided in the repo.

**Comment by bhattg** [comment]
To ensure I followed properly, in my case, I want to find the semantic similarity between the two instructions. Therefore, I'd be feeding something like -- ["Retrieve semantically similar text", "###Instruction: {some prompt} \n. ###Response {some response}"]? 

One more question, how sensitive are LLM2vec model for the instructions? Say if I had used "Encode the following for semantic search" instead of "Retrieve semantically similar text" would it drastically change the results? 

Thanks!
